main.c

```diff
@@ -90,6 +95,69 @@ static void process_share_deprecated(struct entry_settings *opts, const char *op
 	}
 }
 
+/* parse_complete_rlim_t: parses an rlim_t value from `value_str`.  Return 0 on
+   success, 1 otherwise. */
+static int parse_complete_rlim_t(rlim_t *value, char const *value_str) {
+	assert(value_str);
+
+	/* no point in accepting an empty string here... */
+	if (*value_str == '\0') {
+		return 1;
+	}
```

Comment:
I wonder if we should interpret the empty string as meaning "keep the current limit as-is". This would mean that for dropping the soft limit only you could do `--limit-<nofile>=:1000` for instance.